### PR TITLE
Thunks (and keys again)

### PR DESCRIPTION
**Pull request by easafe**
Adds support for snabbdom thunks.

Todo:

- [ ] Add tests for keyed and thunks
- [ ] Add examples for keyed and thunks
- [ ] Mention keys and thunks on the manual

### CODE REVIEW

File: src/Flame/Application/PreMount.purs [hunk 1]
```diff
@@ -99,17 +99,21 @@ preMount (QuerySelector selector) application = do
...
-                injectState node@(Node tag nodeData children)
-                        | tag == "html" =
-                                pure <<< Node tag nodeData $
-                                        case DA.findIndex isBody children of
-                                                Nothing -> state : children
-                                                Just index -> unsafePartial (DM.fromJust $ DA.modifyAt index inject children)
-                        | otherwise = pure $ inject node
\ No newline at end of file
+                isBody = case _ of
+                        Node tag _ _ -> tag == "body"
```
Comment (chekoopa) [inline src/Flame/Application/PreMount.purs:103]: `Node "body" _ _ -> true`

Wonders of pattern matching. (:

**Comment by easafe** [comment]
Actually, this isn't quite right. snabbdom [compares the thunk callback](https://github.com/snabbdom/snabbdom/blob/master/src/package/thunk.ts#L39) as well the arguments, so any kind of currying causes the thunk to rerender.

I think it'd be better UX to depend only on the arguments (perhaps add a clearer docstring as well).

File: src/Flame/Renderer/Renderer.js [hunk 2]
```diff
@@ -41,7 +40,51 @@ function runEvent(handler) {
 
 exports.h_ = h;
 
-exports.thunk_ = thunk;
+function copyToThunk(vnode, thunk) {
+    thunk.elm = vnode.elm;
```
Comment (chekoopa) [inline src/Flame/Renderer/Renderer.js:44]: Duplicate assignment (same as line 50)

Comment (easafe) [reply src/Flame/Renderer/Renderer.js:44]: Oddly, this is how snabbdom has it. Guess they didn't have a review :S

**Pull request closed without merging**